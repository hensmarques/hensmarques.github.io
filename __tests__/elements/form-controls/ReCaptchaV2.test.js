import React from 'react';
import { ReCAPTCHAv2 } from '../../../src/elements/form-controls/ReCaptchaV2';

describe('<ReCAPTCHAv2/>', () => {
  const config = {
    reCaptcha: {
      key: '6LeFaFkUAAAAAJVL3BuBTdv9kT2Hk2fJTKlodoXm',
      theme: 'dark',
    }
  };

  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ReCAPTCHAv2 config={config} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('fires onChange callback', () => {
    const onChangeMock = jest.fn();
    const wrapper = shallow(<ReCAPTCHAv2 config={config} onChange={onChangeMock} />);
    wrapper.instance().onChange('foobar');
    expect(onChangeMock).toHaveBeenCalled();
  });
});
