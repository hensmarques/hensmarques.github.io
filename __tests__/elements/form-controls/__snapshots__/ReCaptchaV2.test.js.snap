// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReCAPTCHAv2/> renders and matches the snapshot 1`] = `
<AsyncScriptLoader(ReCAPTCHA)
  onChange={[Function]}
  sitekey="6LeFaFkUAAAAAJVL3BuBTdv9kT2Hk2fJTKlodoXm"
  theme="dark"
/>
`;
